tests/README.md

```diff
@@ -0,0 +1,173 @@
+make test-single-turn-generation-e2e PROMPT_ID=<prompt-id>
+```
+
+The first target just verifies the server is up, and is most convenient in the local setting where you usually already have the server running in background. The second target is mostly used in CI and will always start and stop a server for you.
+
+#### Structure of the `uses_cases.yaml` file
+
+The `generation/uses_cases.yaml` file holds relevant information on how to run the generation / generated agents tests. Each use case has the following parameters:
+
+- `prompt`: the prompt used to generate the agent.
+- `expected_num_turns`: the expected maximum number of turns agent-factory will take to generate the agent. If generation takes more, [this test](https://github.com/mozilla-ai/agent-factory/blob/faa7ee4edcd0ac6791df6bcf451f2b78010fb108/tests/generation/test_single_turn_generation.py#L88) will fail.
+- `expected_execution_time`: the expected maximum execution time agent-factory will take to generate the agent. If generation takes more, [this test](https://github.com/mozilla-ai/agent-factory/blob/faa7ee4edcd0ac6791df6bcf451f2b78010fb108/tests/generation/test_single_turn_generation.py#L74) will fail.
+- `min_successes`, `max_attempts`: these two parameters are used together. As generation results are non-deterministic, we allow tests to fail sometimes so we expect at least a minimum number of generation to be successful out of a maximum number of runs. This is built to end as early as possible: as soon as we reach `min_successes` runs, the test succeeds; if, before that, we get a number of failures greater than `max_attempts`-`min_successes`, the test fails.
+- `requires_mcpd`: if this is `true`, it means that the generated agent is expected to use MCP tools. This information is used so we can run [mcpd](https://github.com/mozilla-ai/mcpd/) first before running integration tests.
+
+### Artifact Tests
+
+These tests statically validate the generated agent artifacts. They can be run as follows:
+
+```bash
+make test-generated-artifacts PROMPT_ID=<prompt-id>
+```
+
+with `prompt-id` as defined before (see "Generation Tests"). As there are no dependencies on external tools or services, the only command that is run by this Makefile target is:
+
+```
+uv run --group tests pytest tests/generated_artifacts/ -m artifact_validation --prompt-id=$(PROMPT_ID) -v
+```
+
+Note the `-m artifact_validation` parameter: this is a *marker* that is used to group tests under the same namespace so you can follow them more easily in CI. Our namespaces are defined in `/pytest.ini`.
+
+### Artifact Tests (integration)
+
+These tests run the target agents that have been created by *single-turn-generation* and verify they behave as expected.
+Some of the MCP tools are mocked so we don't depend on external applications or paid APIs for our tests.
+As most MCP tools are managed by `mcpd`, its deamon needs to be started beforehand and passed the generated
```

Comment:
Instead of "passed the generated `.mcpd.toml` and `secrets.prod.toml` files" here, is it better to just say:

"Set the variables in the `.env` file that is generated by mcpd` and run the daemon"?

Reply:
Thanks for the thorough explanation!